packages/authentication/src/providers/auth-action.provider.ts

```diff
@@ -29,6 +29,7 @@ export class AuthenticateActionProvider implements Provider<AuthenticateFn> {
     readonly getStrategy: Getter<AuthenticationStrategy>,
     @inject.setter(SecurityBindings.USER)
     readonly setCurrentUser: Setter<UserProfile>,
+    @inject.context() private ctx: Context,
```

Comment:
`inject.setter` is a better option. See https://loopback.io/doc/en/lb4/Decorators_inject.html for examples.

Reply:
We already use `@inject.setter` for user:

```ts
@inject.setter(SecurityBindings.USER)
    readonly setCurrentUser: Setter<UserProfile>,
```

Reply:
agreed